src/Engine/GameAction.java

```diff
@@ -1106,4 +1110,128 @@ public XYCoord getTargetLocation()
       return destination;
     }
   } // ~UnitDeleteAction
+
+  public static class UnitProduceAction implements GameAction
+  {
+    final UnitProduce type;
+    final Unit actor;
+    final XYCoord destination;
+    public UnitProduceAction(UnitProduce pType, Unit unit)
+    {
+      type = pType;
+      actor = unit;
+      destination = new XYCoord(unit.x, unit.y);
+    }
+
+    @Override
+    public GameEventQueue getEvents(MapMaster gameMap)
+    {
+      GameEventQueue eventSequence = new GameEventQueue();
```

Comment:
Are there no preconditions? Funds, unit type, etc?

Reply:
`UnitProduceEvent` should fail properly on its own if we don't have the funds, and unit type is defined... by the production event type, so we can't really check it meaningfully.